mcc/odin/odin/mock_rds_client_test.go

```diff
@@ -16,6 +16,30 @@ type mockRDSClient struct {
 	dbSnapshots          []*rds.DBSnapshot
 }
 
+// AddSnapshots add a list of snapshots to the mock, both in the full
+// list and in the per instance map.
+func (m *mockRDSClient) AddSnapshots(
+	snapshots []*rds.DBSnapshot,
+) {
+	m.dbSnapshots = []*rds.DBSnapshot{}
+	for _, snapshot := range snapshots {
+		m.AddSnapshot(snapshot)
+	}
+}
+
+// AddSnapshot add a new snapshot to the mock, both in the full list
+// and the in the per instance map.
+func (m *mockRDSClient) AddSnapshot(
+	snapshot *rds.DBSnapshot,
+) {
+	m.dbSnapshots = append(m.dbSnapshots, snapshot)
+	id := *snapshot.DBInstanceIdentifier
+	if _, ok := m.dbInstanceSnapshots[id]; !ok {
```

Comment:
This a very Go construct. This could be seen like the following:
```
_, ok := m.dbInstanceSnapshots[id]
if !ok {
...
}```
being the only difference that these variables would be scoped only within the if.
I would mention that the map resolution always returns the value obtained and the success, this one is a bool. In that case, I'm discarding the value got, because I just want to asses if the key exists... so I can handle this case.